### PR TITLE
WIP: issue #415 Add Go CLI to OpenWhisk - Runs Against Whisk Automation

### PR DESCRIPTION
**Pull request by dubeejw**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -38,3 +38,6 @@ node_modules
 
 # Vagrant
 /tools/vagrant/.vagrant
+
+# Webstorm
+**/.idea/
```
Comment (markusthoemmes) [inline .gitignore]: This should probably just be `.idea`. Make sure to remove already commited files from the index, it won't update itself if you ignore files post-commit.

**Comment by csantanapr** [comment]
tracking #415

**Comment by dubeejw** [comment]
@csantanapr,@mdeuser, @rabbah, @markusthoemmes, first green Go CLI build!

**Comment by csantanapr** [comment]
@dubeejw @mdeuser hurray 🙌 ✅

**Comment by dubeejw** [comment]
@rabbah, we would like to start getting pieces of the Go CLI into OpenWhisk. What would be the process to make that happen?

**Comment by rabbah** [comment]
If it's additive (not replacing python CLI), we can do that - in fact we will want them both to co-exist before cutting over: we can next introduce a feature flag to use either of the two version then remove the python and flag.

I would suggest preparing the commit if ready (squash and clean history).
I haven't tracked all the code changes but can start reviewing the code and dependencies in the meantime. @csantanapr @markusthoemmes @psuter additional reviewers?

**Comment by csantanapr** [comment]
I agree with @rabbah let's keep both python-cli for a while until we feel confident to remove the phython code to it's own repo it might become the openwhisk-client-sdk-python

@dubeejw Go ahead and get the PR ready (squash, clean history and remove WIP from title)
Assigned to me when ready.

I will start the review tomorrow. if @markusthoemmes and @psuter can help review that will awesome.

**Comment by dubeejw** [comment]
@rabbah, @csantanapr, I'll work on squashing, and the additive issues tonight. As far as the review goes, I hope it is not too strict as the Go CLI is not finished.

**Comment by dubeejw** [comment]
@csantanapr, @rabbah, @markusthoemmes, @psuter, I'm going to leave this pull request open as it runs the Go CLI against the Whisk automation. Please refer to pull request #604 for what we want merged ASAP. That request only adds the go-cli folder in the tools directory, nothing more.

**Pull request closed without merging**